### PR TITLE
[bugfix] fix megatron opsd with top-k

### PR DESCRIPTION
**Pull request by hjh0119**

### CODE REVIEW

File: swift/megatron/trainers/gkd_trainer.py [hunk 1]
```diff
@@ -658,38 +657,44 @@ def loss_func(self,
...
                 student_logits=student_logits,
-                teacher_logits=teacher_logits,
+                teacher_logits=teacher_output.full_logits,
                 labels=labels,
                 beta=self.beta,
-                teacher_topk_logprobs=teacher_api_logprobs,
-                teacher_topk_indices=teacher_api_indices,
+                teacher_topk_logprobs=teacher_output.topk_logprobs,
+                teacher_topk_indices=teacher_output.topk_indices,
             )
```
[the remark this answers is not in the document]
Comment (hjh0119) [reply swift/megatron/trainers/gkd_trainer.py:700]: labels are shifted in `_prepare_batch` method
